src/XMakeBuildEngine/Instance/ProjectInstance.cs

```diff
@@ -2242,7 +2242,7 @@ private void Initialize(ProjectRootElement xml, IDictionary<string, string> glob
 
             if (NativeMethodsShared.IsUnixLike && !_globalProperties.Contains("MSBuildFrameworkToolsRoot"))
             {
-                _globalProperties.Set(ProjectPropertyInstance.Create("MSBuildFrameworkToolsRoot", NativeMethodsShared.FrameworkBasePath, false));
+                _globalProperties.Set(ProjectPropertyInstance.Create("MSBuildFrameworkToolsRoot", NativeMethodsShared.FrameworkBasePath + '/', false));
```

Comment:
Seems like now that you're doing all that extra work in ToolsetReader to set all the other properties, that would be a better place for this than shoving it into the global properties here.